routing/result_interpretation.go

```diff
@@ -216,6 +217,12 @@ func (i *interpretedResult) processPaymentOutcomeFinal(
 		// deliberately. What to penalize?
 		i.finalFailureReason = &reasonIncorrectDetails
 
+	case *lnwire.FailMPPTimeout:
```

Comment:
i don't think we need to handle this quite yet because the router never sends MPP payments. we only need this if we send HTLCs smaller than `total_amt_msat` which isn't the case rn. that will allow us to defer final judgement on how to penalize as well

Reply:
There is the crowd source case? The idea of doing this now is that we at least don't blacklist our destination.

Btw, marking all the other hops as success can still happen for mpptimeout.

Reply:
yes good point, discussed more offline we still want it since sendtoroute can feed into mission control and bc other implementations will send this error. 

longer term I’m not sure if we need to penalize for mpp_timeout. the router or sendtorroute user should try again until they feel like giving up. so I think this is good as is?

Reply:
As it is now in this pr, we don't penalize for mpp_timeout. we just abort the payment. that should indeed change in the future.

I see that the scucess marking was indeed already in.

Reply:
Also not sure about penalizing for mpp timeout, since we're failing a payment based on another payment not arriving. Difficult to attribute blame there 🤔 

As is, just updated to attribute success to all hops except the last one and just take no action on the last hop? I think that's nice because we still get all the successes from the intermediate hops, and then take no action on the one we're unsure about (which is just a single hop, so we aren't losing a large amount of insight).

Reply:
There is also a lingering bug in here: https://github.com/lightningnetwork/lnd/pull/3863

Reply:
Ah I was wondering about that!